internal/component/local/file_match/watch.go

```diff
@@ -38,7 +39,13 @@ func (w *watch) getPaths() ([]discovery.Target, error) {
 		}
 		fi, err := os.Stat(abs)
 		if err != nil {
-			level.Error(w.log).Log("msg", "error getting os stat", "path", abs, "err", err)
+			// On some filesystems we can get errors accessing the discovered paths. Don't log these as errors.
+			// local.file_match will retry on the next sync period if the access is blocked temporarily only.
+			if utils.IsEphemeralOrFileClosed(err) {
+				level.Debug(w.log).Log("msg", "I/O error when getting os stat", "path", abs, "err", err)
```

Comment:
What if we consistently hit an ephemeral error on every retry? Should we have a timeout that logs it as an error after a while?

Reply:
I think this'd be quite more convoluted for this specific use case (I'm not sure if it'd get stuck in one of these ephemeral errors or keep the closed status for an extended period of time).

Reply:
Then we would not be picking that specific file. Still can be debugged with debug logging. I don't believe that right now we need to go into handling it further.